cmd/kubeadm/app/util/crypto/crypto.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crypto
+
+import (
+	"github.com/pkg/errors"
+
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+)
+
+// createRandBytes returns a cryptographically secure slice of random bytes with a given size
+func createRandBytes(size uint32) ([]byte, error) {
+	bytes := make([]byte, size)
+	if _, err := rand.Read(bytes); err != nil {
+		return nil, err
+	}
+	return bytes, nil
+}
+
+// humanReadableBytes returns a slice of ASCII characters between 33 and 126.
+func humanReadableBytes() ([]byte, int) {
+	result := ""
+	for i := 33; i < 127; i++ {
+		result += string(i)
+	}
+	return []byte(result), len(result)
+}
+
+// CreateHumanReadableRandBytes returns a cryptographically secure slice of random human readable bytes with a given size
+func CreateHumanReadableRandBytes(size uint32) ([]byte, error) {
+	humanReadableBytes, len := humanReadableBytes()
+	bytes, err := createRandBytes(size)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, b := range bytes {
+		bytes[i] = humanReadableBytes[b%byte(len)]
```

Comment:
I'm wondering if we could use `base64.URLEncoding.EncodeToString` given the raw `createRandBytes` instead of performing a module operation here?

Reply:
I has considered this option too, but I'm in doubt about the UX.
@fabriziopandini @neolit123 wdyt?

Reply:
i don't think we need `CreateHumanReadableRandBytes` and `humanReadableBytes`.
to create the random bytes we should use `createRandBytes(size)` and to convert these bytes to a human readable string we should use:
https://golang.org/pkg/encoding/hex/#EncodeToString

```
package main

import (
	"fmt"
	"encoding/hex"
)

func main() {
	b := []byte{32, 64, 123, 253}
	fmt.Println(hex.EncodeToString(b))
	// 20407bfd
}
```

Reply:
Nice!